docs/lcgs/working-examples/Tic_Tac_Toe/tic_tac_toe.lcgs

```diff
@@ -0,0 +1,118 @@
+    || ((pos12 == NOUGHT || nought.play12) && (pos22 == NOUGHT || nought.play22) && (pos32 == NOUGHT || nought.play32))
+    // Noughts on top row
+    || ((pos13 == NOUGHT || nought.play13) && (pos23 == NOUGHT || nought.play23) && (pos33 == NOUGHT || nought.play33))
+    // Noughts on left column
+    || ((pos11 == NOUGHT || nought.play11) && (pos12 == NOUGHT || nought.play12) && (pos13 == NOUGHT || nought.play13))
+    // Noughts on middle column
+    || ((pos21 == NOUGHT || nought.play21) && (pos22 == NOUGHT || nought.play22) && (pos23 == NOUGHT || nought.play23))
+    // Noughts on right column
+    || ((pos31 == NOUGHT || nought.play31) && (pos32 == NOUGHT || nought.play32) && (pos33 == NOUGHT || nought.play33))
+    // Noughts on diagonal /
+    || ((pos11 == NOUGHT || nought.play11) && (pos22 == NOUGHT || nought.play22) && (pos33 == NOUGHT || nought.play33))
+    // Noughts on diagonal \
+    || ((pos13 == NOUGHT || nought.play13) && (pos22 == NOUGHT || nought.play22) && (pos31 == NOUGHT || nought.play31))
+) + CROSS * (
+    // Crosses on bottom row
+    ((pos11 == CROSS || cross.play11) && (pos21 == CROSS || cross.play21) && (pos31 == CROSS || cross.play31))
+    // Crosses on middle row
+    || ((pos12 == CROSS || cross.play12) && (pos22 == CROSS || cross.play22) && (pos32 == CROSS || cross.play32))
+    // Crosses on top row
+    || ((pos13 == CROSS || cross.play13) && (pos23 == CROSS || cross.play23) && (pos33 == CROSS || cross.play33))
+    // Crosses on left column
+    || ((pos11 == CROSS || cross.play11) && (pos12 == CROSS || cross.play12) && (pos13 == CROSS || cross.play13))
+    // Crosses on middle column
+    || ((pos21 == CROSS || cross.play21) && (pos22 == CROSS || cross.play22) && (pos23 == CROSS || cross.play23))
+    // Crosses on right column
+    || ((pos31 == CROSS || cross.play31) && (pos32 == CROSS || cross.play32) && (pos33 == CROSS || cross.play33))
+    // Crosses on diagonal /
+    || ((pos11 == CROSS || cross.play11) && (pos22 == CROSS || cross.play22) && (pos33 == CROSS || cross.play33))
+    // Crosses on diagonal \
+    || ((pos13 == CROSS || cross.play13) && (pos22 == CROSS || cross.play22) && (pos31 == CROSS || cross.play31))
+);
+
+template tictactoe
+    [wait] turn != SHAPE || winner || board_full;
+    [play11] !board_full && !winner && turn == SHAPE && pos11 == EMPTY;
+    [play12] !board_full && !winner && turn == SHAPE && pos12 == EMPTY;
+    [play13] !board_full && !winner && turn == SHAPE && pos13 == EMPTY;
+    [play21] !board_full && !winner && turn == SHAPE && pos21 == EMPTY;
+    [play22] !board_full && !winner && turn == SHAPE && pos22 == EMPTY;
+    [play23] !board_full && !winner && turn == SHAPE && pos23 == EMPTY;
+    [play31] !board_full && !winner && turn == SHAPE && pos31 == EMPTY;
+    [play32] !board_full && !winner && turn == SHAPE && pos32 == EMPTY;
+    [play33] !board_full && !winner && turn == SHAPE && pos33 == EMPTY;
+endtemplate
+
+label tie = board_full && winner == EMPTY;
```

Comment:
In this case we have a tie, when board not full and we have a winner. Which seems strange to me.

Reply:
I don't understand. The game is tied if
* the board is full (no empty positions), and
* there's no winner (no one got three of their shapes on a row)

Reply:
I go confused by precedens. I read it like `(board_full && winner) == EMPTY`. Now i see that is not the case. Then no problems :)